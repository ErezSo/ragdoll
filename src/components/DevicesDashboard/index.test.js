import React from "react";
import { shallow } from "enzyme";
import Dashboard from "./";

describe("Dashboard", () => {
  let wrapper;
  const response = {
    data: [
      {
        name: "rotation_rate_gamma",
        unit: "deg/s",
        value: 0,
        timestamp: 1528878066506,
        active: true
      }
    ]
  };
  beforeEach(() => {
    fetch.mockResponseOnce(JSON.stringify(response));
    wrapper = shallow(<Dashboard />);
  });

  it("renders", () => {
    expect(wrapper).toHaveLength(1);
  });

  it("renders correctly", () => {
    expect(wrapper.debug()).toMatchSnapshot();
  });
});
