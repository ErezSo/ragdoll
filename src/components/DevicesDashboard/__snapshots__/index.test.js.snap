// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard renders correctly 1`] = `
"<div style={{...}}>
  <Panel devices={{...}} handleSearch={[Function]} />
  <DevicesList devices={{...}} searchTerm=\\"\\" toggleDevice={[Function]} />
</div>"
`;
